plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/completion/DAICompletionEngine.java

```diff
@@ -50,13 +50,27 @@ List<DAICompletionResponse> performQueryCompletion(
      * Do query completion in a session
      */
     @NotNull
+    @Deprecated
     List<DAICompletionResponse> performQueryCompletion(
         @NotNull DBRProgressMonitor monitor,
         @NotNull DAICompletionContext context,
         @NotNull DAICompletionSession session,
         @NotNull IAIFormatter formatter
     ) throws DBException;
 
+    /**
+     * Do query completion in a session
+     */
+    @NotNull
+    List<DAICompletionResponse> performQueryCompletion(
+        @NotNull DBRProgressMonitor monitor,
+        @NotNull DAICompletionContext context,
+        @NotNull DAICompletionSession session,
+        @NotNull IAIFormatter formatter,
+        boolean allMessages
```

Comment:
I'd call it `includeAssistantMessages` or `includeResponseMessages`, but even these are not really descriptive.